src/LinAlg/hiopMatrixMDS.hpp

```diff
@@ -332,29 +336,36 @@ class hiopMatrixSymBlockDiagMDS : public hiopMatrix
     assert(false && "not needed / implemented");
   }
 
-  virtual void addMatrix(double alpha, const hiopMatrix& X)
+  virtual void addMatrix(double alpha, const hiopMatrix& X_in)
   {
-    const hiopMatrixSymBlockDiagMDS* pX=dynamic_cast<const hiopMatrixSymBlockDiagMDS*>(&X);
-    if(pX==NULL) {
-      assert(false && "operation only supported for hiopMatrixMDS left operand");
-    } 
-    mSp->addMatrix(alpha, *pX->mSp);
-    mDe->addMatrix(alpha, *pX->mDe);
+    const hiopMatrixSymBlockDiagMDS& X = dynamic_cast<const hiopMatrixSymBlockDiagMDS&>(X_in);
```

Comment:
@ashermancinelli A helper function or dynamic_cast?

Reply:
@nychiang, I am not sure what is your question here. Could you clarify, please?